client/controller/shared.js

```diff
@@ -24,16 +24,12 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 
 		// On server, only render LoggedOutLayout when logged-out.
 		if ( ! context.isServerSide || ! getCurrentUser( context.store.getState() ) ) {
-			let redirectUri;
-			if ( context.isServerSide ) {
-				redirectUri = `${ context.protocol }://${ context.host }${ context.originalUrl }`;
-			}
 			context.layout = (
 				<LayoutComponent
 					store={ store }
 					primary={ primary }
 					secondary={ secondary }
-					redirectUri={ redirectUri }
+					redirectUri={ context.isServerSide ? context.originalUrl : null }
```

Comment:
Do we need the `context.isServerSide ?` check?

Reply:
Good question. Likely not, but let me see...